spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -1463,6 +1449,176 @@ public void testSnapshotProperty() {
     assertThat(table.currentSnapshot().summary()).containsKeys(commitMetricsKeys);
   }
 
+  @Test
+  public void testBinPackRewriterWithSpecificOutputSpec() {
+    Table table = createTable(10);
+    shouldHaveFiles(table, 10);
+    Integer previousSpecId = table.spec().specId();
+    // create multiple partition specs with different commit
+    table.updateSpec().addField(Expressions.truncate("c2", 2)).commit();
+    table.updateSpec().addField(Expressions.bucket("c3", 2)).commit();
+
+    /*
+    This is how the table.specs() look like
+     specId :: PartitionSpec
+      0::[]
+      1::[
+        1000: c2_trunc_2: truncate[2](2)
+      ]
+      2::[
+        1000: c2_trunc_2: truncate[2](2)
+        1001: c3_bucket_2: bucket[2](3)
+      ]
+     */
+    // rewrite to the truncate("c2", 2)
+    int outputSpecId = table.specs().size() - 2;
```

Comment:
why not just use previous spec id? (Note we should probably have a write to unpartitioned spec id test")